packages/neuron-wallet/tests/services/address.test.ts

```diff
@@ -707,5 +712,107 @@ describe('integration tests for AddressService', () => {
         expect(stubbedAddressDbChangedSubjectNext).toHaveBeenCalledTimes(1)
       })
     })
+
+    describe('getPrivateKeyByAddress', () => {
+      const walletService = WalletService.getInstance()
+      const mnemonic = 'tank planet champion pottery together intact quick police asset flower sudden question'
+      const password = '1234abc~'
+      const message = 'Hello World'
+      let address = 'ckb1qzda0cr08m85hc8jlnfp3zer7xulejywt49kt2rr0vthywaa50xwsqwe5xyvcxv95l22x8c5ra8tkc0jgxhvrqsda3p3k'
+
+      let walletID = ''
+
+      beforeAll(() => {
+        jest.setTimeout(15000)
+
+        walletService.clearAll()
+
+        const seed = hd.mnemonic.mnemonicToSeedSync(mnemonic)
+        const masterKeychain = hd.Keychain.fromSeed(seed)
+        const extendedKey = new hd.ExtendedPrivateKey(
+          bytes.hexify(masterKeychain.privateKey),
+          bytes.hexify(masterKeychain.chainCode)
+        )
+
+        const keystore = hd.Keystore.create(extendedKey, password)
+
+        const accountKeychain = masterKeychain.derivePath(hd.AccountExtendedPublicKey.ckbAccountPath)
+        const accountExtendedPublicKey = new hd.AccountExtendedPublicKey(
+          bytes.hexify(accountKeychain.publicKey),
+          bytes.hexify(accountKeychain.chainCode)
+        )
+
+        const wallet = walletService.create({
+          id: '',
+          name: 'Test Wallet',
+          extendedKey: accountExtendedPublicKey.serialize(),
+          keystore,
+        })
+        walletID = wallet.id
+      })
+
+      it('hdWallet address', async () => {
+        const addresses = await walletService.get(walletID).checkAndGenerateAddresses(false, 5, 5)
+        if (addresses) {
+          const crypto = require('crypto')
+          const randomIndex = crypto.randomInt(addresses.length)
+          const obj = addresses[randomIndex]
+          address = obj.address
+        }
+
```

Comment:
I guess `getAddressesByWalletId` can be mocked. Then it's unnecessary to generate addresses.